tests/unit/generator/test_generator.py

```diff
@@ -246,6 +249,86 @@ def test_get_filename_with_proto_and_sub():
     ) == 'bar/types/baz/bacon.py'
 
 
+def test_parse_sample_paths(fs):
+    for fpath in [
+            'sample.yaml',
+            'sampledir/sample.yaml',
+            'other_sampledir/sample.yaml',
+    ]:
+        fs.create_file(
+            fpath,
+            contents=dedent(
+                '''
+                ---
+                type: com.google.api.codegen.SampleConfigProto
+                config_schema_version: 1.2.0
+                samples:
+                - service: google.cloud.language.v1.LanguageService
+                '''
+            )
+        )
+
+    opts = options.Options.build(
+        ("gapic_samples=sample.yaml,"
+         "gapic_samples=sampledir/,"
+         "gapic_samples=other_sampledir"))
+
+    expected_configs = (
+        'sample.yaml',
+        'sampledir/sample.yaml',
+        'other_sampledir/sample.yaml',
+    )
+
+    assert opts.sample_configs == expected_configs
+
+
+@mock.patch(
+    'gapic.samplegen.samplegen.generate_sample',
+    return_value=('sample.py', '')
+)
+@mock.patch(
+    'gapic.samplegen.samplegen.generate_manifest',
+    return_value=('manifest.yaml', yaml.Doc([]))
+)
+def test_samplegen_end_to_end(mock_generate_sample, mock_generate_manifest, fs):
+    fs.create_file(
+        'samples.yaml',
+        contents=dedent(
+            '''
+            ---
+            type: com.google.api.codegen.SampleConfigProto
+            config_schema_version: 1.2.0
+            samples:
+            - service: google.cloud.language.v1.LanguageService
+            '''
+        )
+    )
+
+    g = generator.Generator(
+        options.Options.build(
+            (
+                'gapic_samples=samples.yaml,'
+                'gapic_sample_out_dir=api/{api}/version/{version}/samples'
+            )
+        )
+    )
+    api_schema = make_api(naming=naming.Naming(name='Mollusc', version='v6'))
+    actual_response = g.get_response(api_schema)
+    expected_response = CodeGeneratorResponse(
+        file=[
+            CodeGeneratorResponse.File(
+                name='api/Mollusc/version/v6/samples/sample.py',
+                content="\n"
+            ),
+            CodeGeneratorResponse.File(
+                name="api/Mollusc/version/v6/samples/manifest.yaml",
+                content="---\n"
```

Comment:
there's not sample ID here?

Reply:
No, because actually generating the samples and the manifest is stubbed out. This test only verifies the high level logic of
1) searching for sample configs,
2) iterating over the samples and generating one sample file per config, and
3) generating a manifest for all the sample files.

The _contents_ of those generated files are outside the scope of the test.

Un-stubbing sample and manifest generation would involve constructing an api schema and some sample configs. That functionality is tested elsewhere, and I didn't want to clutter this test with unnecessary code and data.